codalab/apps/web/static/js/worksheet/worksheet_interface.js

```diff
@@ -157,9 +168,41 @@ var Worksheet = React.createClass({
         } else {
           // Go into edit mode.
           this.setState({editMode: editMode});  // Needs to be before focusing
-          this.setState({activeComponent: 'textarea'});
-          // TODO: set cursor intelligently rather than just leaving it at the beginning.
-          $("#raw-textarea").focus();
+          $("#worksheet-editor").focus();
+        }
+    },
+    componentDidUpdate: function(props,state,root) {
+        try {
+            if (this.state.editMode && !this.state.editorEnabled) {
+                this.setState({editorEnabled: true});
+                var editor = ace.edit('worksheet-editor');
+                editor.$blockScrolling = Infinity;
+                editor.session.setUseWrapMode(false);
+                editor.setShowPrintMargin(false);
+                editor.session.setMode('ace/mode/markdown');
+                editor.commands.addCommand({
+                    name: 'save',
+                    bindKey: {win: 'Ctrl-Enter', mac: 'Command-Enter'},
+                    exec: function(editor) {
+                        this.toggleEditMode();
+                    }.bind(this),
+                    readOnly: true
+                });
+                editor.focus();
+
+                var index = this.state.focusIndex + ',' + this.state.subFocusIndex;
+                var defaultIndex = this.state.focusIndex + ',' + '-1';
+                var cursorRowPosition = this.state.ws.info.interpreted_raw_map[index] || this.state.ws.info.interpreted_raw_map[defaultIndex];
+                if (cursorRowPosition === undefined) {
+                    console.error("cannot find this element in the interpreted_raw_map");
```

Comment:
Use single quotes, capitalize first letter, print out more information about the indices